db/db_impl/db_impl_open.cc

```diff
@@ -298,6 +298,11 @@ Status DBImpl::ValidateOptions(const DBOptions& db_options) {
         "writes in direct IO require writable_file_max_buffer_size > 0");
   }
 
+  if (db_options.compaction_readahead_size == 0) {
+    return Status::InvalidArgument(
```

Comment:
Won't this cause any existing options files to fail validation?  Should the "Sanitize" code also be changed to say that if the value is 0, set it to the default?

Reply:
good point - let me think more about it